lib/AST/ASTPrinter.cpp

```diff
@@ -2831,16 +2831,16 @@ class TypePrinter : public TypeVisitor<TypePrinter> {
         Printer << ", ";
       const TupleTypeElt &TD = Fields[i];
       Type EltType = TD.getType();
-      if (auto *IOT = EltType->getAs<InOutType>()) {
-        Printer << "inout ";
-        EltType = IOT->getObjectType();
-      }
+
 
       if (TD.hasName()) {
         Printer.printName(TD.getName(), PrintNameContext::FunctionParameter);
         Printer << ": ";
       }
-
+      if (auto *IOT = EltType->getAs<InOutType>()) {
+        Printer << "inout ";
+        EltType = IOT->getObjectType();
+      }
       if (TD.isVararg()) {
```

Comment:
Another one, the special case for inout should hopefully be possible to drop entirely.